### PR TITLE
Bagstore ingest

### PR DESCRIPTION
**Pull request by rvanheest**
fixes EASY-1127 (bagstore ingest with version-1-deposit)

#### When applied it will
* add a `storage-type` parameter that can either be `dark-archive` (the traditional archive) or `bag-store` (the new archival storage)
* add parameters for the bag-store
* group parameters in `application.properties` such that it is more human-readable
* change the `IngestFlowSettings` creation according to https://github.com/DANS-KNAW/easy-ingest-flow/pull/60
* make sure the `jobMonitor` starts immediately rather than after a couple of seconds (that's the difference between [interval](http://reactivex.io/rxscala/scaladoc/index.html#rx.lang.scala.Observable$@interval(duration:scala.concurrent.duration.Duration):rx.lang.scala.Observable[Long]) and [interval](http://reactivex.io/rxscala/scaladoc/index.html#rx.lang.scala.Observable$@interval(initialDelay:scala.concurrent.duration.Duration,period:scala.concurrent.duration.Duration):rx.lang.scala.Observable[Long])

#### Where should the reviewer @DANS-KNAW/easy start?
`EasyIngestDispatcher.getIngestFlowSettings`

#### How should this be manually tested?
**STILL ON MY TODO LIST**

#### related pull requests on github
repo                      | PR
----------------- | -----------------
easy-dtap            | [PR#53](https://github.com/DANS-KNAW/easy-dtap/pull/53)
easy-ingest-flow | [PR#60](https://github.com/DANS-KNAW/easy-ingest-flow/pull/60)

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/easy/ingest_dispatcher/EasyIngestDispatcher.scala [hunk 1]
```diff
@@ -133,9 +134,25 @@ abstract class EasyIngestDispatcher(implicit props: PropertiesConfiguration) {
 
   def getIngestFlowSettings(deposit: Deposit): IngestFlowSettings = {
     IngestFlowSettings(
-      storageUser = props.getString("storage.user"),
-      storagePassword = props.getString("storage.password"),
-      storageServiceUrl = new URL(props.getString("storage.service-url")),
+      storageSettings = props.getString("storage.type") match {
+        case "dark-archive" => DarkArchiveSettings(
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/ingest_dispatcher/EasyIngestDispatcher.scala]: @rvanheest : where is `DarkArchiveSettings`? Could it be that you forgot to push it to your fork? I cannot find it anywhere. Please update asap.

Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/ingest_dispatcher/EasyIngestDispatcher.scala]: `DarkArchiveSettings` class is in easy-ingest-flow, `nl.dans.knaw.easy.ingest_flow.package.scala`

Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/ingest_dispatcher/EasyIngestDispatcher.scala]: Okay, my bad. Not looking hard enough.

**Comment by janvanmansum** [comment]
Created and merged a new PR based on this one, solving merge conflicts.

**Pull request closed without merging**